set.mm

```diff
@@ -24785,72 +24785,107 @@ range over class builders (implicitly in the case of defined class terms
-     treats the theory of classes as an extralogical extension to our logic and
-     set theory axioms.  He calls the construction ` { y | ph } ` a "class
-     term".
-
-     While the three class definitions ~ df-clab , ~ df-cleq , and ~ df-clel
-     are eliminable and conservative and thus meet the requirements for sound
-     definitions, they are technically axioms in that they do not satisfy the
-     requirements for the current definition checker.  The proofs of
-     conservativity require external justification that is beyond the scope of
-     the definition checker.
-
-     In particular, note that ~ df-clab defines a class abstraction if only if
-     a class abstraction is completely determined by which elements it
-     contains, which is the content of the axiom of extensionality ~ ax-ext .
-     Therefore, ~ df-clab can be considered a definition only in systems that
-     can prove ~ ax-ext (and the necessary first-order logic).
-
-     For a general discussion of the theory of classes, see
-     ~ mmset.html#class .  (Contributed by NM, 26-May-1993.) $)
+  $( Define class abstractions, that is, classes of the form ` { y | ph } ` ,
+     which is read "the class of sets ` y ` such that ` ph ( y ) ` ".
+
+     A few remarks are in order:
+
+     1.  The axiomatic statement ~ df-clab does not define the class
+     abstraction ` { y | ph } ` itself, that is, it does not have the form
+     ` |- { y | ph } = ... ` that a standard definition should have (for a good
+     reason: equality itself has not yet been defined or axiomatized for class
+     abstractions; it is defined later in ~ df-cleq ).  Instead, ~ df-clab has
+     the form ` |- ( x e. { y | ph } <-> ... ) ` , meaning that it only defines
+     what it means for a setvar to be a member of a class abstraction.  As a
+     consequence, one can say that ~ df-clab defines class abstractions if and
+     only if a class abstraction is completely determined by which elements
+     belong to it, which is the content of the axiom of extensionality
+     ~ ax-ext .  Therefore, ~ df-clab can be considered a definition only in
+     systems that can prove ~ ax-ext (and the necessary first-order logic).
+
+     2.  As in all definitions, the LHS (definiendum) has no disjoint variable
+     conditions.  In particular, the setvar variables ` x ` and ` y ` need not
+     be distinct, and the formula ` ph ` may depend on both ` x ` and ` y ` .
+     This is necessary, as with all definitions, since if there was for
+     instance a disjoint variable condition on ` x , y ` , then one could not
+     do anything with expressions like ` x e. { x | ph } ` which are sometimes
+     useful to shorten proofs (because of ~ abid ).  Most often, however, ` x `
+     is fresh in ` x e. { y | ph } ` and ` y ` is free in ` ph ` .
+
+     3.  Remark 1 stresses that ~ df-clab does not have the standard form of a
+     definition for a class, but one could be lead to think it has the standard
```

Comment:
"could be lead" should be "could be led"

Reply:
done